### PR TITLE
Only run unit and integration tests on oldest and latest python versions for macos

### LINKED ISSUE #294: Only run unit and integration tests on oldest and latest python versions for macos
### Problem Description

_As an engineer, I'd like to see the results of CI workflows quickly so that I cna make changes or merge PRs more efficiently._

Github only allows 5 concurrent mac machines, while it allows 20 total machines. For this reason, our CI builds often get bottlenecked at the mac builds.

### Expected behavior

- Only run the unit and integration workflows on python 3.8 and 3.12 for Macos

### PR DESCRIPTION
**Pull request by R-Palazzo**
CU-86b07h9tx
Resolve #294

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86b07h9tx SDGym - Only run unit and integration tests on oldest and latest python versions for macos #294](https://app.clickup.com/t/86b07h9tx)

File: pyproject.toml [hunk 1]
```diff
@@ -26,29 +26,29 @@ dependencies = [
...
     "torch>=2.0.0;python_version>='3.10' and python_version<'3.12'",
     "torch>=2.2.0;python_version>='3.12'",
     'tqdm>=4.29',
     'XlsxWriter>=1.2.8',
-    'rdt>=1.11.0',
-    'sdmetrics>=0.14.0',
-    'sdv>=1.12.0',
+    'rdt  @ git+https://github.com/sdv-dev/RDT@main',
+    'sdmetrics @ git+https://github.com/sdv-dev/SDMetrics@main',
+    'sdv @ git+https://github.com/sdv-dev/SDV@main',
```
Comment (amontanez24) [inline pyproject.toml:51]: why is this needed?

Comment (R-Palazzo) [reply pyproject.toml:51]: Otherwise, the builds were crashing:
https://github.com/sdv-dev/SDGym/actions/runs/8893490555/job/24420381562

I can try without SDV pointing to main, according to the error message it seems there is conflict between rdt, sdmetrics and sklearn.

**Review by lajohn4747 — APPROVED** [review]
LGTM - Should we make an issue to update dependencies to not point to main for the next release

**Comment by amontanez24** [comment]
> LGTM - Should we make an issue to update dependencies to not point to main for the next release

I think we should wait to merge this one

**Review by amontanez24 — APPROVED** [review]
